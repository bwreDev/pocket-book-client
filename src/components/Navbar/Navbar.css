.navbar-list {
  display: flex;
  justify-content: space-around;
  list-style: none;
  margin: 0;
  padding: 0;
  padding-top: 30px;
}

.navbar-list li a {
  color: black;
  text-decoration: none;
}

.navbar-list li a:hover {
  color: #78d853;
}
