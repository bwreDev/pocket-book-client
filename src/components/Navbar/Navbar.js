import React, { Component } from 'react';
import { Link } from 'react-router-dom';
//import TokenService from '../../services/token-services';
//import IdleService from '../../services/idle-service';
import './Navbar.css';

export default class Navbar extends Component {
  /*
  state = {
    user: null,
  };

  handleLogoutClick = () => {
    TokenService.clearAuthToken();
    TokenService.clearCallbackBeforeExpiry();
    IdleService.unRegisterIdleResets();
  };

  renderLogoutLink() {
    return (
      <>
        <li>
          <Link to='/user'>Profile</Link>
        </li>
        <li>
          <Link to='/input'>Add Input</Link>
        </li>
        <li>
          <Link to='/' onClick={this.handleLogoutClick}>
            Logout
          </Link>
        </li>
      </>
    );
  }

  renderLoginLink() {
    return (
      <>
        <li>
          <Link to='/register'>Register</Link>
        </li>
        <li>
          <Link to='/login'>Log in</Link>
        </li>
      </>
    );
  }
  */

  render() {
    return (
      <nav role='navigation' className='navbar'>
        <ul className='navbar-list'>
          <li>
            <Link to='/'>Home</Link>
          </li>
          <li>
            <Link to='/login'>Login</Link>
          </li>
          <li>
            <Link to='/register'>Register</Link>
          </li>
          <li>
            <Link to='/user'>Profile</Link>
          </li>
          <li>
            <Link to='/input'>Input</Link>
          </li>
          {/*
          TokenService.hasAuthToken()
            ? this.renderLogoutLink()
          : this.renderLoginLink()
          */}
        </ul>
      </nav>
    );
  }
}
