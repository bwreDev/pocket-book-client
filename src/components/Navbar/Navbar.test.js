import React from 'react';
import ReactDOM from 'react-dom';
import Navbar from './Navbar';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

it('should render without error', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
