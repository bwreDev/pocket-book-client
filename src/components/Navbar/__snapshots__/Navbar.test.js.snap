// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<nav
  className="navbar"
  role="navigation"
>
  <ul
    className="navbar-list"
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </li>
    <li>
      <a
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </li>
  </ul>
</nav>
`;
